draft-normington-jsonpath-latest.xml

```diff
@@ -202,24 +202,32 @@ root-selector = %x24               ; $ selects document root node
                   <name>Syntax</name>
                   <t>A dot child matcher has a key known as a dot child name or a single asterisk (<tt>*</tt>).
                   </t>
-                  <t>A dot child name corresponds to a key in a JSON object, without the surrounding double quotes (<tt>"</tt>).
+                  <t>A dot child name corresponds to a name in a JSON object. It can be unquoted or surrounded with quotes, either double (<tt>"</tt>) or single (<tt>'</tt>).
                   </t>
                   <sourcecode type="abnf">
 matcher = dot-child             ; see below for alternatives
 dot-child = %x2E dot-child-name / ; .&lt;dot-child-name&gt;
             %x2E %x2A             ; .*
-dot-child-name = TBD
+dot-child-name = union-child / ; "key" or 'key' (see below)
```

Comment:
Well, what I see now is: the majority of implementations support the syntax itself (so we should support it, too, I believe). But there's a difference in a way of decoding the key: some implementations consider quotes as non-special part of the ID ("raw" approach) and some "unquote" the string. The following links show that there's no clear "majority" in the approach, but I think that "unquote" approach is much less surprising than "raw" one.

https://cburgmer.github.io/json-path-comparison/results/dot_notation_with_double_quotes.html
https://cburgmer.github.io/json-path-comparison/results/dot_notation_with_single_quotes.html

Reply:
It's tricky because either way could be surprising. If we adopt the "unquote" approach, then why bother having bracket child notation at all? It would really be superfluous. I'd find that surprising.

Also, both Gössner's implementations adopt the "raw" approach and I feel that we should probably use Gössner to break any "ties" now that we are going to merge into an internet draft which will include him as an author.

Reply:
> If we adopt the "unquote" approach, then why bother having bracket child notation at all?

Interesting point, very good. But on the other hand, we have no "bracket child" anymore: brackets define union now (and the former "bracket child" is just a special case of singular union) - that's the main reason for having them, I believe.

I feel that if both interpretations are surprising, then we should probably forbid the quotes in dot-child-name at all, bringing some existing queries as a sacrifice to clarity. What do you think?

Reply:
That seems reasonable to me and it's consistent with the definition of `unquoted-key` which already prohibits single and double quotes from appearing anywhere.